tensorboard/plugins/graph/tf_graph_controls/tf-graph-controls.ts

```diff
@@ -583,184 +610,194 @@ class TfGraphControls extends LegacyElementMixin(PolymerElement) {
-                x1="0%"
-                y1="0%"
-                x2="100%"
-                y2="0%"
-              >
-                <stop
-                  class="start"
-                  offset="0%"
-                  stop-color$="[[_currentGradientParams.startColor]]"
-                ></stop>
-                <stop
-                  class="end"
-                  offset="100%"
-                  stop-color$="[[_currentGradientParams.endColor]]"
-                ></stop>
-              </linearGradient>
-            </defs>
-            <rect
-              x="0"
-              y="0"
-              width="135"
+    </div>
+    <div class="legend-holder">
+      <div class="legend-toolbar" on-click="_toggleLegendOpen">
+        <span class="toggle-legend-text">Legend</span>
+        <paper-icon-button
+          icon="[[_getToggleLegendIcon(_legendOpened)]]"
+          class="toggle-legend-button"
+        >
+        </paper-icon-button>
+      </div>
+      <iron-collapse opened="[[_legendOpened]]" class="legend-content">
+        <!-- Color-mode-specific legend items -->
+        <div>
+          <template is="dom-if" if="[[_isGradientColoring(stats, colorBy)]]">
+            <svg
+              width="140"
               height="20"
-              fill="url(#linearGradient)"
-              stroke="black"
-            ></rect>
-          </svg>
-          <div class="domainValues color-text">
-            <div class="domainStart">[[_currentGradientParams.minValue]]</div>
-            <div class="domainEnd">[[_currentGradientParams.maxValue]]</div>
-          </div>
-          <br style="clear: both" />
-          <div>Devices included in stats:</div>
-          <div class="deviceList">
-            <template is="dom-repeat" items="[[_currentDevices]]">
-              <div class="color-legend-row devices-checkbox">
-                <span
-                  ><input
-                    type="checkbox"
-                    value$="[[item.device]]"
-                    checked$="[[item.used]]"
-                    on-click="_deviceCheckboxClicked"
-                /></span>
-                <span>[[item.suffix]]</span>
-                <template is="dom-if" if="[[item.ignoredMsg]]">
-                  <paper-icon-button
-                    icon="help"
-                    class="help-icon"
-                  ></paper-icon-button>
-                  <paper-tooltip position="right" offset="0" animation-delay="0"
-                    >[[item.ignoredMsg]]</paper-tooltip
-                  >
-                </template>
-              </div>
-            </template>
-          </div>
-        </template>
-        <template is="dom-if" if="[[_equals(colorBy, 'structure')]]">
-          <div class="color-text">
-            <div class="color-legend-row">
-              <span class="label"> colors </span>
-              <span class="color-legend-value">same substructure</span>
+              style="margin: 0 5px"
```

Comment:
let's correct it by putting this on the CSS?

Reply:
Done.